Modules/OptimismMainModule.php

```diff
@@ -27,5 +27,31 @@ function initialize()
         {
             return '0';
         };
+
+        // Handles
+        $this->handles_implemented = true;
+        $this->handles_regex = '/(.*)\.box/';
+        $this->api_get_handle = function($handle)
+        {
+            if (!preg_match($this->handles_regex, $handle))
+                return null;
+
+            require_once __DIR__ . '/../Engine/Crypto/Keccak.php';
+
+            $hash = $this->ens_name_to_hash($handle);
+
+            if (is_null($hash) || $hash === '')
+                return null;
+
+            $address = $this->ens_get_data_from_resolver('0xf97aac6c8dbaebcb54ff166d79706e3af7a813c8', $hash, '0x3b3b57de', -40);
+
+            if ($address === '0000000000000000000000000000000000000000') // try to call ownerOf on .box NFT contract
+                $address = $this->ens_get_data_from_resolver('0xBB7B805B257d7C76CA9435B3ffe780355E4C4B17', $hash, '0x6352211e', -40);
```

Comment:
@Har01d This resolves domains that didn't setup `ENS Forward Resolution`
As domain's NFT ID is derived from the ENS `namehash` of the domain, we can get the owner of the NFT and the address we get is the resolution of the domain.
I've checked a bunch of domains and didn't found the exception from this rule.

If there's a possibility to set the resolution address different from NFT receiver address, this should be done ENS way, and our first call to the resolver address wouldn't be there.
I believe we should keep this approach to resolve domains without `ENS Forward Resolution`